atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueCacheSnapshotImpl.java

```diff
@@ -30,13 +30,18 @@
     Set<TableReference> enabledTables();
 
     @Override
-    default boolean isWatched(TableReference tableReference) {
-        return enabledTables().contains(tableReference);
+    default Optional<CacheEntry> getValue(CellReference tableAndCell) {
+        return values().get(tableAndCell).toJavaOptional();
     }
 
     @Override
-    default Optional<CacheEntry> getValue(CellReference tableAndCell) {
-        return values().get(tableAndCell).toJavaOptional();
+    default boolean isUnlocked(CellReference tableAndCell) {
+        return getValue(tableAndCell).map(CacheEntry::isUnlocked).orElse(true);
```

Comment:
No, this is that the default assumes the value is unlocked unless proven otherwise (nothing to do with the table being watched IIRC).

Reply:
But that is incorrect. It is only unlocked because the semantics of being watched means it's unlocked unless explicitly locked. We do not have such a guarantee for unwatched tables. I know I'm splitting hairs though since it literally does not matter for unwatched tables, but we should just be defensive and return false as the default

Reply:
Ah, I think I see your point. I think we need to do this:

1. If table is unwatched, return false;
2. If table is watched, return false if the value is locked, else return true (even if the value is absent)

I wanted to ensure that we return true if no value is there (because that means we just don't have a cached value or a lock taken out), but didn't check whether the table was watched.